src/Uno.UI/UI/Xaml/UIElement.cs

```diff
@@ -48,8 +48,8 @@ public partial class UIElement : DependencyObject, IXUidProvider, IUIElement
 		private static readonly Type[] _bringIntoViewRequestedArgs = new[] { typeof(BringIntoViewRequestedEventArgs) };
 
 		private readonly SerialDisposable _clipSubscription = new SerialDisposable();
-		private XamlRoot _xamlRoot = null;
 		private string _uid;
+		private WeakReference<VisualTree> _visualTreeCacheWeakReference;
```

Comment:
It would make sense to have `VisualTree` implement `IWeakReferenceProvider` and rely on the `ManagedWeakReference` to handle this. We'll otherwise create a lot of references to the same object.